### ISSUE TITLE
[Intermittent] Mozilla VPN activation screen flickers when switching location

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/mozilla-vpn-client#5802]
**Important details:** 

* Reproducible on production server as well;

* The steps for reproducing this issue are not always the same;

* Quitting the app seems to resolve the issue;

* The same issue was encountered a few days ago: [+VPN-4012+](https://mozilla-hub.atlassian.net/browse/VPN-4012); 

**Affected versions:**

* 2.13.0 (2.202301272129);

**Tested Platforms:**

* MacOS 12.6.1;

* Windows 11;

**Prerequisites:**

* Have Mozilla VPN app installed and opened;

* Have a Fx account with an active subscription for Mozilla VPN;

* Be signed in;

**Steps to reproduce:**

1. Sign out;
1. Sign back in;
1. Turn ON the VPN;
1. Switch location a few times; 

**Expected result:**

* The activation screen does not have any display issues;

* “VPN Switched Servers” OS notification is displayed only once;

**Actual result:**

* The activation screen flickers when connecting to another server location;

* “VPN Switched Servers” OS notification is displayed 2 or 3 times;

**Notes:**

* See the attached video and logs;

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/VPN-4033)
┆Reporter: Bianca Hidecuti

**Comment by data-sync-user** [comment]
➤ Rebecca Billings commented:

Bianca Hidecuti we’re going to keep this issue for the screen flicker, please file a separate issue for the duplicate notifications. Thanks!

**Comment by data-sync-user** [comment]
➤ Sarah Bird commented:

Some thoughts here: 

  * we’re quickly transitioning to the connect state and back so it’s technically correct
  * but the flickering could be an accessibility problem and is not a good ux
  * we should probably either show the connection state intentionally longer to communicate what’s happening OR not rerender changes so quickly to prevent flickering

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

I was able to reproduce this, using the same 2.13 build, Guardian 1.28, while on prod, after a sign out and back in, at select location change.

When the issue was logged I was not able to reproduce this on my Windows 10 device.

!flicker2.gif|width=670,height=846!

**Comment by data-sync-user** [comment]
➤ Bianca Hidecuti commented:

Rebecca Billings, the duplicate notifications are displayed only when the screen flickers, because the VPN it is actually reconnecting. When the activation screen does not flicker, the OS notification is displayed only once. 

If filling a separate issue for the notifications makes more sense, please let me know! Thank you.

**Comment by data-sync-user** [comment]
➤ Bianca Hidecuti commented:

I am able to reproduce this on Mozilla VPN 2.15.0 (2.202304120927), while using macOS 12.6.1, when switching locations.

See the attached video.

!Screen Recording 2023-04-13 at 16.13.23.mov|width=716,height=1486!

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

I’ve encountered this again today a few times on Windows 10, using Mozilla VPN 2.16.0 (2.202306220247) build.

Attaching video and logs.

!flickflack.gif|width=774,height=959!

[^mozillavpn-2023-6-22.txt]

**Comment by data-sync-user** [comment]
➤ Raluca Monica Ilban commented:

I was able to reproduce the issue today on Android 11 using Mozilla VPN 2.18.0 (2.202309140656)

!screen-20230914-180426.mp4|width=1440,height=2880!
